design-documents/graph-ql/coverage/negotiableQuotes.graphqls

```diff
@@ -1,205 +1,332 @@
-    id: ID! @doc(description: "Unique Identifier of Negotiable Quote Item")
-    product: ProductInterface!
-    qty: Float! @doc(description: "Quantity added")
-    customizable_options: [SelectedCustomizableOption] @doc(description: "custom Option selected")
-    bundle_options: [SelectedBundleOption]! @doc(description: "selected bundle options")
+type NegotiableQuoteComment {
+    id: ID!
+    created_at: String!
+    author: NegotiableQuoteUser!
+    creator_type: NegotiableQuoteCommentCreatorType!
+    value: String! @doc(description: "A simple (non-html) comment submitted by a seller or buyer")
 }
 
-type ConfigurableRequisitionListItem implements NegotiableQuoteItemInterface
-@doc(description: "Negotiable Quote Item Implementation that for Configurable Products") {
-    id: ID! @doc(description:  "Unique Identifier of Negotiable Quote Item")
-    product: ProductInterface!
-    qty: Float! @doc(description: "Quantity added")
-    customizable_options: [SelectedCustomizableOption] @doc(description: "custom Option selected")
-    configurable_options: [SelectedConfigurableOption] @doc(description: "Configurable options selected")
+enum NegotiableQuoteCommentCreatorType {
+    BUYER
+    SELLER
 }
 
-type AttachmentContent @doc(description: "Negotiable quote attachment file") {
-    base64_encoded_data: String!
-    mime_type: String!
+type NegotiableQuote {
+    id: ID!
     name: String!
+    items: [NegotiableQuoteItem!]
+    # Attachment Support is dependent on headless File Upload design
+    # attachments: [AttachmentContent]
+    comments: [NegotiableQuoteComment!]
+    history: [NegotiableQuoteHistoryEntry!]
+    prices: CartPrices
+    buyer: NegotiableQuoteUser!
+    created_at: String @doc(description: "Timestamp indicating when the negotiable quote was created.")
+    updated_at: String @doc(description: "Timestamp indicating when the negotiable quote was updated.")
+    status: NegotiableQuoteStatus!
 }
 
-type NegotiableQuoteComment {
-    id: ID!
-    parent_id: String!
-    author: String!
-    text: String
-    created_at: String
-    attachments: [String]
+# Implementation Note: Using the values from the "Buyer Status" column of the state mapping table in devdocs.
+# These states are identical between storefront/admin, but we name them differently for buyers.
+# https://devdocs.magento.com/guides/v2.4/b2b/negotiable-quote.html#quote-statuses
+enum NegotiableQuoteStatus {
+    SUBMITTED
+    PENDING
+    UPDATED
+    OPEN
+    ORDERED
+    CLOSED
+    DECLINED
+    EXPIRED
+}
+
+type NegotiableQuoteItem @doc(description: "A line item on a Negotiable Quote") {
+    id: ID! @doc(description: "ID of the line item (not product) in a Negotiable Quote")
+    item: CartItemInterface @doc(description: "Product assigned to line item, with selected configurations")
+}
+
+input NegotiableQuoteFilterInput {
+    ids: FilterEqualTypeInput @doc(description: "Filter by Negotiable Quote ID(s)")
```

Comment:
Do we really want to have separate `negotiable_quote(id: ID!)` query considering the ability to use this filter?

Reply:
Yeah, they serve distinct use-cases.

1. An ID lookup query gives a client a clear way to fetch a single entity in the cache. This is useful because we don't have a root `Node` interface like most GraphQL APIs, so it's not immediately clear how to fetch a type by its ID.

2. The filter input is for filtering/searching in a list view.

We got feedback from `pwa-studio` folks with our `products` query before that it made it complicated to just find a way to get the product by ID (one of the most common use-cases for a UI). The more filter we add to `NegotiableQuoteFilterInput`, the more muddled the use-case of the query is (lookup vs search).

Reply:
I'd say it's a similar reason to why SQL has a `=` operator, instead of requiring you to use `LIKE` or `IN` - you often just want to get an entity by ID.